### DISCUSSION TITLE
How to avoid the osd frequent restart?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#11613] [Q&A]
For example:
1. ceph cluster has nine nodes, and one of the nodes has a problem with the network (latency is 4xxx ms)
2. because of  the heartbeat timeout, the mon will mark the osd under the nodes to down.
3. the liveness of the osd will pull the osd up.
4. repeat  above procedure， the osd will frequent restart

if the  osd frequent restart, it will cause of the slow ops or block request. https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/4/html-single/troubleshooting_guide/index#flapping-osds_diag

**Comment by sp98** [comment]
@zhucan You might want to checkout the healthcheck liveness probe for [OSD] container (https://github.com/rook/rook/blob/4dc38d3b0fc4f5ceb44768638c0e630bb88fc9d0/pkg/operator/ceph/cluster/osd/spec.go#L594) and adjust time accordingly.

**Reply by zhucan** [reply]
Is there number of retries to config? @sp98

**Reply by sp98** [reply]
You can try `failureThreshold` configuration option mentioned [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). This is the closest option I can find for the retry logic.

**Reply by zhucan** [reply]
Thanks, I will try it.
